src/NuGet.Core/NuGet.Protocol.Core.v3/Providers/FeedTypeResourceProvider.cs

```diff
@@ -26,18 +26,21 @@ public FeedTypeResourceProvider()
 
             if (source.FeedTypeOverride == FeedType.Undefined)
             {
-                // Check the feed type
-                var feedType = FeedTypeUtility.GetFeedType(source.PackageSource);
-
-                if (feedType == FeedType.FileSystemUnknown)
-                {
-                    // Do not cache unknown folder types
-                    curResource = new FeedTypeResource(FeedType.FileSystemUnknown);
-                }
-                else
+                if (!_feedTypeCache.TryGetValue(source.PackageSource, out curResource))
```

Comment:
This should be done in the GetOrAdd so that it is done at most one time. Potentially this could allow multiple threads to search for the type.

I suggest moving everything inside the undefined if statement into another method, then returning _feedTypeCache.GetOrAdd